backend-tests/tests/test_deployments.py

```diff
@@ -0,0 +1,257 @@
+import pytest
+from datetime import datetime, timedelta
+import time
+import os
+import requests
+
+import api.client
+from api.client import ApiClient
+import api.deviceauth as deviceauth_v1
+import api.useradm as useradm
+import api.inventory as inventory
+from common import User, Device, Authset, Tenant, \
+        create_tenant, create_tenant_user, \
+        make_accepted_devices, clean_mongo, mongo
+
+
+def upload_image(filename, auth_token, description="abc"):
+    image_path_url = api.client.GATEWAY_URL + "/api/management/v1/deployments/artifacts"
+    r = requests.post(
+        image_path_url,
+        verify=False,
+        headers={"Authorization": "Bearer " + auth_token},
+        files=(("description", (None, description)),
+               ("size", (None, str(os.path.getsize(filename)))),
+               ("artifact", (filename, open(filename, 'rb'),
+                             "application/octet-stream"))))
+    assert r.status_code == 201
+
+
+@pytest.fixture(scope='function')
+def setup_deployments_enterprise_test(clean_mongo, nr_devices=100):
+    tenant = create_tenant('acme')
+    user = create_tenant_user('bugs-bunny', tenant)
+    useradmm = ApiClient(useradm.URL_MGMT)
+    devauthd = ApiClient(deviceauth_v1.URL_DEVICES)
+    invm = ApiClient(inventory.URL_MGMT)
+    deploymentclient = ApiClient(
+        api.client.GATEWAY_URL + "/api/management/v1/deployments")
+    # log in user
+    r = useradmm.call('POST', useradm.URL_LOGIN, auth=(user.name, user.pwd))
+    assert r.status_code == 200
+    utoken = r.text
+    # Upload a dummy artifact to the server
+    upload_image("/tests/test-artifact.mender", utoken)
+    # Get the artifact from the server
+    r = deploymentclient.with_auth(utoken).call('GET', '/artifacts')
+    # prepare accepted devices
+    devs = make_accepted_devices(utoken, devauthd, nr_devices,
+                                 tenant.tenant_token)
+    # wait for devices to be provisioned
+    time.sleep(3)
+    r = invm.with_auth(utoken).call(
+        'GET', inventory.URL_DEVICES, qs_params={
+            'per_page': 100
+        })
+    assert r.status_code == 200
+    api_devs = r.json()
+    assert len(api_devs) == 100
+    return (user, tenant, utoken, api_devs)
+
+
+class TestDeploymentsEndpointEnterprise(object):
+    test_cases = [
+        # Base-case
+        ({
+            "name": "first-test-case",
+            "artifact_name": "deployments-phase-testing",
+            "devices": ["is set in test body"],
+        }, {
+            "name": "first-test-case",
+            "artifact_name": "deployments-phase-testing",
+            "device_count": 100
+        }),
+        # Phase, set start time
```

Comment:
remove comments, have more descriptive "name", like `"name": "one phase, with start, no batch"`, and then
`"name": "one phase, with start, with batch"`
`"name": "> one phase, first no start, last no batch"`
`"name": "> one phase, first with start, last no batch"`, etc (just examples)